frontend/app/src/main/java/com/example/speechbuddy/compose/resetpassword/ResetPasswordScreen.kt

```diff
@@ -65,49 +69,43 @@ fun ResetPasswordScreen(
 
                 // Password Text Field
                 TextFieldUi(
-                    value = password.value,
-                    onValueChange = { password.value = it },
+                    value = viewModel.passwordInput,
+                    onValueChange = { viewModel.setPassword(it) },
                     label = { Text(stringResource(id = R.string.new_password_field)) },
-                    supportingText = { Text(stringResource(id = R.string.false_new_password)) },
-                    isError = false,
-                    isValid = false,
+                    supportingText = {
+                        if (isPasswordError) {
+                            Text(stringResource(id = uiState.error!!.messageId))
+                        }
+                    },
+                    isError = isError,
                     isHidden = true
                 )
 
                 // Password Check Text Field
                 TextFieldUi(
-                    value = passwordCheck.value,
-                    onValueChange = { passwordCheck.value = it },
+                    value = viewModel.passwordCheckInput,
+                    onValueChange = { viewModel.setPasswordCheck(it) },
                     label = { Text(stringResource(id = R.string.new_password_check_field)) },
-                    supportingText = { Text(stringResource(id = R.string.false_new_password_check)) },
-                    isError = false,
-                    isValid = false,
+                    supportingText = {
+                        if (isPasswordCheckError) {
+                            Text(stringResource(id = uiState.error!!.messageId))
+                        }
+                    },
+                    isError = isError,
                     isHidden = true
                 )
 
-                Spacer(modifier = Modifier.height(15.dp))
+                Spacer(modifier = Modifier.height(20.dp))
 
                 // Set password Button
                 ButtonUi(
                     text = stringResource(id = R.string.reset_password_next),
-                    onClick = onNextClick,
+                    onClick = { viewModel.resetPassword(navController) },
                     isError = false,
                     isEnabled = true,
```

Comment:
```suggestion
                    isError = isError,
```

isEnabled는 어차피 default value가 true여서 삭제했습니다